staging/src/k8s.io/apimachinery/pkg/watch/mux.go

```diff
@@ -196,7 +193,10 @@ func (m *Broadcaster) Action(action EventType, obj runtime.Object) {
 // have received the data yet as it can remain sitting in the buffered
 // channel.
 func (m *Broadcaster) Shutdown() {
-	close(m.incoming)
+	m.blockQueue(func() {
```

Comment:
this change takes the serial close and then wait and makes the the close async, not guaranteed before the wait.

Reply:
> this change takes the serial close and then wait and makes the the close async, not guaranteed before the wait.

Reading through, I think the change still works.

Reply:
😄